flink-rpc/flink-rpc-akka/src/test/java/org/apache/flink/runtime/rpc/akka/RobustActorSystemTest.java

```diff
@@ -84,6 +84,34 @@ void testUncaughtExceptionHandlerFromActor() {
         assertThat(uncaughtException).isSameAs(error);
     }
 
+    @Test
+    void testHonorClassloadingErrorBeforeShutdown() {
+        robustActorSystem
+                .uncaughtExceptionHandler()
+                .uncaughtException(Thread.currentThread(), new NoClassDefFoundError(""));
+
+        assertThat(testingUncaughtExceptionHandler.findUncaughtExceptionNow()).isPresent();
+    }
+
+    @Test
+    void testIgnoreClassloadingErrorAfterShutdown() {
```

Comment:
nit: this test could be split up in two test methods (or through a parameterized test), e.g.:

```
    @ParameterizedTest
    @ValueSource(classes = {NoClassDefFoundError.class, ClassNotFoundException.class})
    void testIgnoreClassloadingErrorAfterShutdown(Class<? extends Throwable> clazz)
            throws InstantiationException, IllegalAccessException {
        // wait for termination
        robustActorSystem.terminate();
        robustActorSystem.getWhenTerminated().toCompletableFuture().join();

        robustActorSystem
                .uncaughtExceptionHandler()
                .uncaughtException(Thread.currentThread(), clazz.newInstance());

        assertThat(testingUncaughtExceptionHandler.findUncaughtExceptionNow()).isEmpty();
    }
```